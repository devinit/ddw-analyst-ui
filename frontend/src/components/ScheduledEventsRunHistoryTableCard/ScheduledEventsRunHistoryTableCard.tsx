import React, { FunctionComponent } from 'react';
import { Card } from 'react-bootstrap';
import { Dimmer, Loader } from 'semantic-ui-react';
import { ScheduledEventsRunHistoryTable } from '../ScheduledEventsRunHistoryTable';

export const ScheduledEventsRunHistoryTableCard: FunctionComponent = () => {
  return (
    <React.Fragment>
      <Dimmer inverted>
        <Loader content="Loading" />
      </Dimmer>
      <Card className="col-md-12">
        <Card.Header className="card-header-rose card-header-icon">
          <h4 className="card-title">Update FTS Run History</h4>
        </Card.Header>
        <Card.Body>
          <ScheduledEventsRunHistoryTable />
        </Card.Body>
      </Card>
    </React.Fragment>
  );
};
