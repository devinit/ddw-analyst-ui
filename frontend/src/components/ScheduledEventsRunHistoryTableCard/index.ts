export * from './ScheduledEventsRunHistoryTableCard';
export * from './utils';
