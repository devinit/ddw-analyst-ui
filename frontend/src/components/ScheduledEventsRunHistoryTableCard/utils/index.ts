import axios from 'axios';
import * as localForage from 'localforage';
import { ScheduledEventRunHistory } from '../../../types/scheduledEvents';
import { api, localForageKeys } from '../../../utils';
import { LIMIT } from '../../ScheduledEventsTableCard';

const BASEPATH = api.routes.FETCH_RUN_INSTANCES;

export const fetchRunHistory = async (eventId: number): Promise<any> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const headers = {
    'Content-Type': 'application/json',
    Authorization: `token ${token}`,
  };

  return await axios(`${BASEPATH.replace('{id}', eventId.toString())}`, { headers });
};

export const getScheduledEventsByPage = (
  currentPage: number,
  data: ScheduledEventRunHistory[],
): ScheduledEventRunHistory[] => {
  const begin = (currentPage - 1) * 5;
  const end = begin + LIMIT;

  return data.slice(begin, end);
};
