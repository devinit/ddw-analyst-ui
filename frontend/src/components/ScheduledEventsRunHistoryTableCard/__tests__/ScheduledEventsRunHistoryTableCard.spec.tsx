/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import { cleanup, render, waitFor, act } from '@testing-library/react';
import 'jest-styled-components';
import React from 'react';
import { ScheduledEventsRunHistoryTableCard } from '../ScheduledEventsRunHistoryTableCard';
import { AxiosResponse } from 'axios';
import localforage from 'localforage';

jest.mock('localforage');
const data = {
  results: [
    {
      scheduled_event: 1,
      start_at: new Date('2020-01-01').toISOString(),
      status: 'pending',
    },
  ],
  count: 1,
};

const axiosResponse: AxiosResponse = {
  data: data,
  status: 200,
  statusText: 'OK',
  config: {},
  headers: {},
};

jest.mock('axios', () => ({
  __esModule: true,
  get: jest.fn(() => Promise.resolve(axiosResponse)),
  default: jest.fn(() => Promise.resolve(axiosResponse)),
}));

describe('<ScheduledEventsRunHistoryTableCard', () => {
  beforeEach(async () => {
    jest.clearAllMocks();
    await localforage.setItem('localForageKeys.API_KEY', '0695904053');
  });
  const event = {
    id: 1,
    name: 'event 1',
    description: null,
    enabled: true,
    interval: 2,
    interval_type: 'secs',
    repeat: true,
    start_date: new Date('2020-01-01').toISOString(),
  };
  afterEach(cleanup);
  afterEach(() => jest.clearAllMocks());

  it('should match snapshot', async () => {
    await act(async () => {
      const { container } = render(<ScheduledEventsRunHistoryTableCard event={event} />);
      expect(container).toMatchSnapshot();
    });
  });

  it('should fetch data per page on page load', async () => {
    await act(async () => {
      const { getByText } = render(<ScheduledEventsRunHistoryTableCard event={event} />);
      await waitFor(() => expect(getByText('Showing 1 to 1 of 1')));
    });
  });
});
