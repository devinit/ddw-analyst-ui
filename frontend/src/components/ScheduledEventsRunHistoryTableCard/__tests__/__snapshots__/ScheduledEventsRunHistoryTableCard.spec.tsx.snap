// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScheduledEventsRunHistoryTableCard should match snapshot 1`] = `
<div>
  <div
    class="ui active transition visible inverted dimmer"
    style="display: flex;"
  >
    <div
      class="content"
    >
      <div
        class="ui text loader"
      >
        Loading
      </div>
    </div>
  </div>
  <div
    class="col-md-12 card"
  >
    <div
      class="card-header-danger card-header-icon card-header"
    >
      <h4
        class="card-title"
      >
        event 1
         Run History
      </h4>
    </div>
    <div
      class="card-body"
    >
      <div
        class="dropdown"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown-toggle btn btn-danger btn-sm"
          id="status-filter"
          type="button"
        >
          Filter By Status = All
        </button>
      </div>
      <div
        class="table-responsive"
      >
        <table
          class="table table-striped"
        >
          <thead>
            <tr>
              <th>
                Status
              </th>
              <th>
                Started
              </th>
              <th>
                Ended
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      No Data
    </div>
  </div>
</div>
`;
